middle_end/flambda/simplify/simplify_rec_info_expr.ml

```diff
@@ -0,0 +1,134 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                       Pierre Chambart, OCamlPro                        *)
+(*           Mark Shinwell and Leo White, Jane Street Europe              *)
+(*                                                                        *)
+(*   Copyright 2013--2019 OCamlPro SAS                                    *)
+(*   Copyright 2014--2021 Jane Street Group LLC                           *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+[@@@ocaml.warning "+a-30-40-41-42"]
+
+module DA = Downwards_acc
+module DE = Downwards_env
+module T = Flambda_type
+
+let compute_succ
+      ~(depth : int Or_infinity.t)
+      ~(unrolling : Rec_info_expr.Unrolling_state.t) =
+  let depth : int Or_infinity.t =
+    match depth with
+    | Finite n -> Finite (n+1)
+    | Infinity -> Infinity
+  in
+  let unrolling =
+    match unrolling with
+    | Not_unrolling | Unrolling { remaining_depth = 0 } | Do_not_unroll ->
```

Comment:
Let's make this a bit more robust: if `remaining_depth <= 0` then return `unrolling`.